disperser/dataapi/queried_operators_handlers.go

```diff
@@ -56,7 +56,38 @@ func (s *server) getDeregisteredOperatorForDays(ctx context.Context, days int32)
 	return DeregisteredOperatorMetadata, nil
 }
 
-func processOperatorOnlineCheck(deregisteredOperatorState *IndexedDeregisteredOperatorState, operatorOnlineStatusresultsChan chan<- *DeregisteredOperatorMetadata, logger logging.Logger) {
+func (s *server) getRegisteredOperatorForDays(ctx context.Context, days int32) ([]*QueriedStateOperatorMetadata, error) {
+	// Track time taken to get deregistered operators
+	startTime := time.Now()
+
+	indexedRegisteredOperatorState, err := s.subgraphClient.QueryIndexedOperatorsWithStateForTimeWindow(ctx, days, Registered)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert the map to a slice.
+	operators := indexedRegisteredOperatorState.Operators
+
+	operatorOnlineStatusresultsChan = make(chan *QueriedStateOperatorMetadata, len(operators))
+	processOperatorOnlineCheck(indexedRegisteredOperatorState, operatorOnlineStatusresultsChan, s.logger)
+
+	// Collect results of work done
+	RegisteredOperatorMetadata := make([]*QueriedStateOperatorMetadata, 0, len(operators))
+	for range operators {
+		metadata := <-operatorOnlineStatusresultsChan
+		RegisteredOperatorMetadata = append(RegisteredOperatorMetadata, metadata)
+	}
+
+	// Log the time taken
+	s.logger.Info("Time taken to get registered operators for days", "duration", time.Since(startTime))
+	sort.Slice(RegisteredOperatorMetadata, func(i, j int) bool {
+		return RegisteredOperatorMetadata[i].BlockNumber < RegisteredOperatorMetadata[j].BlockNumber
+	})
+
+	return RegisteredOperatorMetadata, nil
+}
+
+func processOperatorOnlineCheck(deregisteredOperatorState *IndexedQueriedOperatorInfo, operatorOnlineStatusresultsChan chan<- *QueriedStateOperatorMetadata, logger logging.Logger) {
```

Comment:
Should the name of `deregisteredOperatorState` be different since it now accepts both registed and deregistered?